graphrag/vector_stores/couchbasedb.py

```diff
@@ -0,0 +1,193 @@
+                        )
+                except Exception as e:
+                    logger.error(
+                        f"Error occurred while loading document {doc['id']}: {str(e)}"
+                    )
+
+            logger.info(
+                f"Successfully loaded {successful_loads} out of {len(batch)} documents"
+            )
+        else:
+            logger.warning("No valid documents to load")
+
+    def similarity_search_by_text(
+        self, text: str, text_embedder: TextEmbedder, k: int = 10, **kwargs: Any
+    ) -> list[VectorStoreSearchResult]:
+        """Perform ANN search by text."""
+        logger.info(f"Performing similarity search by text with k={k}")
+        query_embedding = text_embedder(text)
+        if query_embedding:
+            return self.similarity_search_by_vector(query_embedding, k)
+        logger.warning("Failed to generate embedding for the query text")
+        return []
+
+    def similarity_search_by_vector(
+        self, query_embedding: list[float], k: int = 10, **kwargs: Any
+    ) -> list[VectorStoreSearchResult]:
+        """Perform ANN search by vector."""
+        logger.info(f"Performing similarity search by vector with k={k}")
+
+        search_req = SearchRequest.create(
+            VectorSearch.from_vector_query(
+                VectorQuery(
+                    self.embedding_key,
+                    query_embedding,
+                    k,
+                )
+            )
+        )
+
+        if self.scoped_index:
+            search_iter = self.scope.search(
+                self.index_name,
+                search_req,
+                SearchOptions(
+                    limit=k,
+                    fields=["*"],
+                ),
+            )
+        else:
+            search_iter = self.db_connection.search(
+                index=self.index_name,
+                request=search_req,
+                options=SearchOptions(limit=k, fields=["*"]),
+            )
+
+        results = []
+        for row in search_iter.rows():
+            text = row.fields.pop(self.text_key, "")
+            metadata = self._format_metadata(row.fields)
+            score = row.score
+            doc = VectorStoreDocument(
+                id=row.id,
+                text=text,
+                vector=row.fields.get(self.embedding_key),
+                attributes=metadata,
+            )
+            results.append(VectorStoreSearchResult(document=doc, score=score))
+
+        logger.info(f"Found {len(results)} results in similarity search by vector")
+        return results
+
+    def filter_by_id(self, include_ids: list[str] | list[int]) -> Any:
+        """Build a query filter to filter documents by id."""
+        id_filter = ",".join([f"{id!s}" for id in include_ids])
+        logger.debug(f"Created filter by ID: {id_filter}")
+        return f"search.in(id, '{id_filter}', ',')"
```

Comment:
Are you using this?

Reply:
its not necessary for the graphrag demo but for the vector store. similar to the other ones